### DISCUSSION TITLE
User pr first contribution status

### DISCUSSION THREAD

**Discussion by acrobat** [community/community#24511] [Apps, API and Webhooks]
<p>Is it possible to check through the api ( for example the pr endpoint) if the pr is the first contribution of this user on the repository?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/acrobat">@acrobat</a>,</p>
<p>Thank you for being here! I suppose you could query the “First-time contributor” labels, using: <a href="https://developer.github.com/v3/issues/labels/" rel="nofollow noopener">https://developer.github.com/v3/issues/labels/</a> to start. </p>

**Comment by acrobat** [comment]
<p>I’m indeed talking about the label in the PR header</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/6a5b782978c64bf752003760b489a34db152506c.png" data-download-href="/uploads/short-url/faSAdliYZ12miwHhi4gMoMRGrco.png?dl=1" title="Screenshot 2020-04-27 at 20.30.13.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1374857/181085119-fd3aec62-bdaa-43ce-a2cf-eabb7d8b3e0e.png" alt="Screenshot 2020-04-27 at 20.30.13.png" data-base62-sha1="faSAdliYZ12miwHhi4gMoMRGrco" width="690" height="106" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/6a5b782978c64bf752003760b489a34db152506c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-04-27 at 20.30.13.png</span><span class="informations">766×118 3.06 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But the label api doesn’t return this label for this pullrequest</p>

**Comment by acrobat** [comment]
<p><a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a> any update on how to get this info through the api?</p>

**Comment by francisfuzz** [comment, marked as answer]
<p>:wave: <a class="mention" href="/acrobat">@acrobat</a> - The GraphQL API has a <a href="https://docs.github.com/en/graphql/reference/enums#commentauthorassociation"><code>CommentAuthorAssociation</code></a> enum available. <a href="https://graphql.github.io/graphql-spec/June2018/#sec-Enums">Enums</a> represent possible sets of values for a field.</p>
<p>Let’s use this issue as an example, where you’re currently listed as a <code>Collaborator</code>:</p>
<aside class="onebox githubissue">
  <header class="source">
      <a href="https://github.com/KnpLabs/php-github-api/issues/873" target="_blank">github.com/KnpLabs/php-github-api</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/KnpLabs/php-github-api/issues/873" target="_blank">Replacing the "Delete reactions" endpoint</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2020-05-10" data-time="18:13:29" data-timezone="UTC">06:13PM - 10 May 20 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/acrobat" target="_blank">
          <img alt="acrobat" src="https://user-images.githubusercontent.com/15894826/181085121-19a8d819-cd75-4382-8dc7-a33e8cc1228e.png" class="onebox-avatar-inline" width="20" height="20">
          acrobat
        </a>
      </div>
    </div>
  </div>
</div>

<div class="github-row">
  <p class="github-content">See https://developer.github.com/changes/2020-02-26-new-delete-reactions-endpoints/</p>
</div>

<div class="labels">
    <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">feature-request</span>
</div>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You can use <a href="https://developer.github.com/v4/explorer/">our GitHub GraphQL API Explorer</a> to run the following query:</p>
<pre><code class="lang-auto">query {
  repository(owner: "KnpLabs", name: "php-github-api") {
    issue(number: 873) {
      author {
        login
      }
      authorAssociation
    }
  }
}
</code></pre>
<p>The API will respond with the appropriate information:</p>
<pre><code class="lang-json">{
  "data": {
    "repository": {
      "issue": {
        "author": {
          "login": "acrobat"
        },
        "authorAssociation": "COLLABORATOR"
      }
    }
  }
}
</code></pre>
<p>Hope this helps! :v:</p>

**Comment by acrobat** [comment]
<p>Thanks a lot <a class="mention" href="/francisfuzz">@francisfuzz</a>! This does indeed solve my question!</p>
